package/kedro_viz/server.py

```diff
@@ -213,31 +244,66 @@ def commands():
     type=click.Path(dir_okay=False, writable=True),
     help="Path to save the pipeline JSON file",
 )
-def viz(host, port, browser, load_file, save_file):
+@click.option(
+    "--pipeline",
+    type=str,
+    default=None,
+    help="Name of the modular pipeline to run."
+    "If not set, the default pipeline is visualized",
+)
+@click.option(
+    "--env",
+    "-e",
+    type=str,
+    default=None,
+    multiple=False,
+    envvar="KEDRO_ENV",
+    help="Kedro configuration environment. If not specified, "
+    "catalog config in `local` will be used",
+)
+def viz(host, port, browser, load_file, save_file, pipeline, env):
     """Visualize the pipeline using kedroviz."""
-    _call_viz(host, port, browser, load_file, save_file)
-
-
-def _call_viz(host=None, port=None, browser=None, load_file=None, save_file=None):
+    try:
+        _call_viz(host, port, browser, load_file, save_file, pipeline, env)
+    except KedroCliError:
+        raise
+    except Exception as ex:
+        raise KedroCliError(str(ex))
+
+
+# pylint: disable=too-many-arguments
+def _call_viz(
+    host=None,
+    port=None,
+    browser=None,
+    load_file=None,
+    save_file=None,
+    pipeline_name=None,
+    env=None,
+):
     global data  # pylint: disable=global-statement,invalid-name
 
     if load_file:
-        data = json.loads(Path(load_file).read_text())
-        for key in ["nodes", "edges", "tags"]:
-            if key not in data:
-                click.echo("Invalid file, top level key '{}' not found.".format(key))
-                sys.exit(1)
+        data = _load_from_file(load_file)
     else:
         try:
-            pipeline = get_project_context("create_pipeline")()
-            catalog = get_project_context("create_catalog")(None)
-        except KedroContextError:
-            raise KedroCliError(
-                "Could not find a Kedro project root. "
-                "You can run `kedro viz` by either providing `--load-file` flag with a JSON file "
-                "path for your pipeline, or if the current working directory is "
-                "the root of a Kedro project."
-            )
+            context = get_project_context("context", env=env)
+        # In Kedro<0.15.0, `get_project_context` does not have **kwargs,
+        # so `env=env` fails with TypeError. Or it fails with `context` keyword with KeyError.
+        except (KeyError, TypeError):
+            # Kedro <0.15.0
+            try:
```

Comment:
Can you refactor the logic for Kedro 0.14.* in another helper function please?
So many levels of nesting are usually error prone and not good for the readability either.